spec/unit/AndroidManifest.spec.js

```diff
@@ -0,0 +1,311 @@
+            expect(manifest.getMinSdkVersion()).toBe(1);
+        });
+    });
+
+    describe('maxSdkVersion', () => {
+        it('should get maxSdkVersion', () => {
+            expect(manifest.getMaxSdkVersion()).toBe(MAX_SDK_VERSION);
+        });
+
+        it('should set maxSdkVersion', () => {
+            manifest.setMaxSdkVersion(1);
+            expect(manifest.getMaxSdkVersion()).toBe(1);
+        });
+
+        it('should create the uses-sdk node if it does not exist when setting maxSdkVersion', () => {
+            const NO_SDK_MANIFEST = DEFAULT_MANIFEST.replace(/<uses-sdk.*\/>/, '');
+            createTempManifestFile(NO_SDK_MANIFEST);
+            manifest = new AndroidManifest(manifestPath);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).toBe(null);
+
+            manifest.setMaxSdkVersion(1);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).not.toBe(null);
+            expect(manifest.getMaxSdkVersion()).toBe(1);
+        });
+    });
+
+    describe('targetSdkVersion', () => {
+        it('should get targetSdkVersion', () => {
+            expect(manifest.getTargetSdkVersion()).toBe(TARGET_SDK_VERSION);
+        });
+
+        it('should set targetSdkVersion', () => {
+            manifest.setTargetSdkVersion(1);
+            expect(manifest.getTargetSdkVersion()).toBe(1);
+        });
+
+        it('should create the uses-sdk node if it does not exist when setting targetSdkVersion', () => {
+            const NO_SDK_MANIFEST = DEFAULT_MANIFEST.replace(/<uses-sdk.*\/>/, '');
+            createTempManifestFile(NO_SDK_MANIFEST);
+            manifest = new AndroidManifest(manifestPath);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).toBe(null);
+
+            manifest.setTargetSdkVersion(1);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).not.toBe(null);
+            expect(manifest.getTargetSdkVersion()).toBe(1);
+        });
+    });
+
+    describe('debuggable', () => {
+        it('should get debuggable', () => {
+            expect(manifest.getDebuggable()).toBe(true);
+        });
+
+        it('should remove debuggable if set to a falsy value', () => {
+            manifest.setDebuggable(false);
+            expect(manifest.doc.getroot().find('./application').attrib['android:debuggable']).toBe(undefined);
+        });
+
+        it('should set debuggable to true', () => {
+            const NO_DEBUGGABLE_MANIFEST = DEFAULT_MANIFEST.replace('android:debuggable="true"', '');
+            createTempManifestFile(NO_DEBUGGABLE_MANIFEST);
+            manifest = new AndroidManifest(manifestPath);
+
+            expect(manifest.getDebuggable()).toBe(false);
+
+            manifest.setDebuggable(true);
+            expect(manifest.getDebuggable()).toBe(true);
+        });
+    });
+
+    describe('write', () => {
+        let fsSpy;
+
+        beforeEach(() => {
+            fsSpy = jasmine.createSpyObj(['writeFileSync']);
```

Comment:
Use `jasmine.createSpyObj('fs', ['writeFileSync'])` or any Jasmine error messages will refer to the spy as `undefined.writeFileSync`